Development Progress - week 02

rayapetkova opened the pull request:
New pull request - week 02

karastoyanov committed:
Test file

karastoyanov committed:
update user data fully implemented

karastoyanov committed:
code refactor

karastoyanov committed:
re-open categories fixed

karastoyanov commented on pyqt5-gui/main_menu.py:
```diff
@@ -286,6 +286,7 @@ def open_update_account():
             reset_button.setFixedWidth(200)
+            reset_button.clicked.connect(lambda: update_user_settings_groupbox.hide())
             reset_button.clicked.connect(lambda: open_update_account())
             
```
dgfsdh

karastoyanov committed:
prevent re-opening any user menu

karastoyanov commented:
With b2db7ce9fab814b2f5c2840fa5d842db0b03e88c major bug fix is solved. After the user opens any of the user menus from the left layout, `.setEnabled(False)` sets the respective button as inactive and prevents it to be called again, which generates multi-layer stacking.

```python
            # Disable the button to avoid calling again the function
                # Not the best approach, but for now it will do
            buttons[-1].setEnabled(False)

            global hide_user_update_settings
            def hide_user_update_settings():
                update_user_settings_groupbox.hide()
                categories_groupbox.show()
                # Activate back the button
                buttons[-1].setEnabled(True)
```

rayapetkova committed:
Updated test images, products function

karastoyanov committed:
najor fixes in products:
 - function open_products() calls products window
 - open_products() inits additional groupbox and replaces it with
   subcategories groupbox in the main_layout

rayapetkova committed:
Added some changes - deleted the groupboxes for every pduct

karastoyanov committed:
products structure updated

rayapetkova committed:
Update: new images, work with database and some changes in subcategories' file

- New images
- Added an item_description in db_handle --> manage_portfolio and changed the query because of the added item description
- Added a button "about" in subcategories.py
- Changed the query on line 231 (added the product_description and product_id to it)
- Made two deques for "products_descriptions" and "products_ids"
- One change on line 247 --> "Sunglasses" to "{cat}"

karastoyanov committed:
qplainttextedit implemented for the products description
